resolvers.py

```diff
@@ -0,0 +1,1979 @@
+        qs = qs.filter(department='Delivery')
+
+        # validate capability_name
+        if str(band_name).lower() == "total":
+            return qs
+        else:
+            qs = qs.filter(band=band_name)
+
+        return qs
+
+
+class TotalBaseDataSourceResolverByBand:
+    @custom_resolver_check_perm
+    def __call__(self, source, info, **kwargs):
+        # params required
+        # 1. band name
+        # 2. type of data (Available, Nominated, Allocation Pending, Total bench)
+
+        band_name = kwargs.get("band_name")
+        bench_type = kwargs.get("bench_type")
+
+        qs_filters = Q(nomination_status__in=bench_type.value)
+
+        if None in bench_type.value:
+            qs_filters = qs_filters | Q(nomination_status__isnull=True)
+
+        qs = models.BaseBenchDataSource.objects.filter(qs_filters)
+        qs = qs.filter(updated_end_date__isnull=False)
+        qs = qs.filter(band__in=BENCH_REPORTS_BAND,
+                       on_bench__in=ON_BENCH, allocation_end_dates__isnull=True)
+        qs = qs.filter(department='Delivery')
+
+        if str(band_name).lower() == "total":
+            return qs
+        else:
+            qs = qs.filter(band=band_name)
+
+        return qs
+
+
+# Cell Details Resolver for column names "Allocation Pending", "Nominated", "Bench", "Deployable Total (Grand Total)"
+# for Overall Bench View By Band
+
+class BaseDataSourceResolverForOverallBenchByBandDeployable:
+    @custom_resolver_check_perm
+    def __call__(self, source, info, **kwargs):
+        # params required
+        # 1. band name
+        # 2. column name (Allocation_Pending, Nominated, Bench, Deployable_Total)
+
+        band_name = kwargs.get("band_name")
+        column_name = kwargs.get("column_name")
+
+        qs_filters = Q(nomination_status__in=column_name.value)
+
+        if None in column_name.value:
+            qs_filters = qs_filters | Q(nomination_status__isnull=True)
+
+        qs = models.BaseBenchDataSource.objects.filter(qs_filters)
+        qs = qs.filter(updated_end_date__isnull=False)
+        qs = qs.filter(band__in=BENCH_REPORTS_BAND,
+                       on_bench__in=ON_BENCH, allocation_end_dates__isnull=True)
+        qs = qs.filter(department='Delivery')
+
+        if str(band_name).lower() == 'total':
+            return qs
+        else:
+            qs = qs.filter(band=band_name)
+
+        return qs
+
+
+# Cell Details Resolver for column names "Resigned", "LOP", "Maternity", "Non Deployable Total (Grand Total)"
+# for Overall Bench View By Band
+
+class BaseDataSourceResolverForOverallBenchByBandNonDeployable:
+    @custom_resolver_check_perm
+    def __call__(self, source, info, **kwargs):
+        # params required
+        # 1. band name
+        # 2. column (resigned, loss of pay, maternity, non_deployable_total)
+
+        band_name = kwargs.get("band_name")
+        column_name = kwargs.get("column_name")
+        today = date.today()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Naive datetime objects are treated by many datetime methods as local times, We recommend that you create aware timestamps with the `now` or `fromtimestamp` functions using the `tzinfo` to attribute to represent times in UTC. Do not use the [datetime.utcnow()](https://docs.python.org/3/library/datetime.html#datetime.datetime.utcnow) because it returns only a `naive` datetime object. 

[Learn more](https://docs.python.org/3/library/datetime.html#aware-and-naive-objects)